user_guide_src/source/changelogs/v4.3.0.rst

```diff
@@ -52,6 +52,7 @@ Changes
     - The ``CodeIgniter\CLI\CommandRunner`` class has been removed due to a change in Spark commands processing.
     - The system route configuration file ``system/Config/Routes.php`` has been removed.
     - The route configuration file ``app/Config/Routes.php`` has been changed. Removed include of system routes configuration file.
+- All union type properties in ``Config`` classes have been typed.
```

Comment:
Union is ambiguous here as we use PHP 7.4, better to use nullable and/or strict scalar types.

Reply:
Oh, thanks. I was completely confused. 
I meant that we cannot use union type, so I want to say *atomic* type.

Reply:
I like "scalar" because it has precedence with PHPStan and docblocks. It's been a long time since I heard "atomic types", we used that term back in C/C++ 😊

Reply:
But I see you sent atomic already, that is fine!

Reply:
I think `array` is not `scalar`.

Reply:
You're right: int, float, string, bool. https://www.php.net/manual/en/function.is-scalar.php

But are arrays atomic? Would "simple" be more clear? Or "non-class"?

Reply:
*atomic* is not the official word in PHP.
But Psalm uses it.
https://psalm.dev/docs/annotating_code/type_syntax/atomic_types/

I just mean it is not union nor intersection.

Reply:
Thanks for the link! That's exactly the kind of thing I was looking for. "Atomic" in C/C++ means something different, having to do with thread-safeness IIRC, but ends up being a similar set of "simpler" types. 

I like what we have here, let's stick with it.